lib/mayaUsd/ufe/UsdAttribute.h

```diff
@@ -152,12 +147,16 @@ class UsdAttribute
 #endif
 #endif
 
+    PXR_NS::UsdPrim          usdPrim() const { return _attrHolder->usdPrim(); }
+    PXR_NS::UsdAttribute     usdAttribute() const { return _attrHolder->usdAttribute(); }
+    PXR_NS::SdfValueTypeName usdAttributeType() const;
+    Ufe::AttributeEnumString::EnumValues getEnumValues() const
+    {
+        return _attrHolder->getEnumValues();
+    }
+
 protected:
-    PXR_NS::UsdPrim fPrim;
-#ifdef UFE_V4_FEATURES_AVAILABLE
-    const Ufe::AttributeDef::ConstPtr fAttrDef;
-#endif
-    PXR_NS::UsdAttribute fUsdAttr;
+    std::unique_ptr<UsdAttributeHolder> _attrHolder;
```

Comment:
Only a single member, which is the attribute holder. All callbacks delegate to the holder implementation.